skimage/filters/thresholding.py

```diff
@@ -726,6 +727,134 @@ def threshold_triangle(image, nbins=256):
     return bin_centers[arg_level]
 
 
+def threshold_multiotsu(image, classes=3, bins=255):
+    """Generates multiple thresholds for an input image.
+    Based on the Multi-Otsu approach by Liao, Chen and Chung.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Grayscale input image.
+    classes : int, optional
+        Number of classes to be thresholded, i.e. the number of resulting
+        regions. Accepts an integer from 2 to 5. Default is 3.
+    bins : int, optional
+        Number of bins used to calculate the histogram. Default is 255.
+
+    Returns
+    -------
+    idx_thresh : array
+        Array containing the threshold values for the desired classes.
+
+    References
+    ----------
+    .. [1] Liao, P-S., Chen, T-S. and Chung, P-C., "A fast algorithm for
+           multilevel thresholding", Journal of Information Science and
+           Engineering 17 (5): 713-727, 2001. Available at:
+           <http://ftp.iis.sinica.edu.tw/JISE/2001/200109_01.pdf>
+    .. [2] Tosa, Y., "Multi-Otsu Threshold", a java plugin for ImageJ.
+           Available at:
+           <http://imagej.net/plugins/download/Multi_OtsuThreshold.java>
+
+    Examples
+    --------
+    >>> from skimage.color import label2rgb
+    >>> from skimage import data
+    >>> image = data.camera()
+    >>> thresh = threshold_multiotsu(image)
+    >>> regions = np.digitize(image, bins=thresh)
+    >>> regions_colorized = label2rgb(regions)
+    """
+    # receiving minimum and maximum values for the image type.
+    type_min, type_max = dtype_limits(image)
+
+    # calculating the histogram and the probability of each gray level.
+    hist, _ = np.histogram(image.ravel(), bins=bins,
+                           range=(type_min, type_max))
+    prob = hist / image.size
+
+    max_sigma = 0
+    momP, momS, var_btwcls = [np.zeros((bins, bins)) for n in range(3)]
```

Comment:
a comment explaining the meaning of these variables would be welcome

Reply:
`momP` and `momS` are moments of something :laughing: They are tables used for calculating the variances between classes (`var_btwcls` itself). I'll re-read the paper to make better comments on that.